admin/load.php

```diff
@@ -336,3 +336,23 @@ function perflab_get_module_data( $module_file ) {
 
 	return $module_data;
 }
+
+/**
+ * Enable all non-experimental modules on plugin activation.
+ *
+ * @since 1.0.0
+ */
+function perflab_activation_hook() {
+	$modules          = perflab_get_modules();
+	$modules_settings = perflab_get_module_settings();
```

Comment:
One crucial thing missing here is that we should only set the option if it has no value yet. For example, if you already used the plugin and later reactivated it, it shouldn't modify any option value previously defined by the user.
```suggestion
function perflab_activation_hook() {
	// Bail if option is already set with any value.
	if ( false !== get_option( PERFLAB_MODULES_SETTING, false ) ) {
		return;
	}

	$modules          = perflab_get_modules();
	$modules_settings = perflab_get_module_settings();
```